### PR TITLE
Fix bugs with MSE and Quantile losses for 2D tensors

### PR DESCRIPTION
**Pull request by shengy90**
Previous implementation of weighted MSE and Quantile is wrong (`torch.sum(squared_loss * sample_weight) / torch.sum(sample_weights)`). See below for explanation.

**Examples:**
```python

weights = torch.tensor([1, 1, 3, 1, 2])

tensor1 = torch.tensor([
        [2., 3., 4., 5., 6., 7.],
        [1., 1., 1., 2., 3., 4.],
        [2., 3., 4., 5., 6., 7.],
        [1., 1., 1., 2., 3., 4.],
        [1., 1., 1., 2., 3., 4.]
])

tensor2 = torch.tensor([
        [1., 1., 1., 2., 3., 4.],
        [1., 1., 1., 2., 3., 4.],
        [1., 1., 1., 2., 3., 4.],
        [1., 1., 1., 2., 3., 4.],
        [1., 1., 1., 2., 3., 4.]
])

# vice-versa for tensor2
tensor1_expanded = torch.tensor([
        [2., 3., 4., 5., 6., 7.],
        [1., 1., 1., 2., 3., 4.],
        [2., 3., 4., 5., 6., 7.],
        [2., 3., 4., 5., 6., 7.],
        [2., 3., 4., 5., 6., 7.],
        [1., 1., 1., 2., 3., 4.],
        [1., 1., 1., 2., 3., 4.],
        [1., 1., 1., 2., 3., 4.]
])
```

**Using F.mse_loss on `tensor1_expanded` and tensor2_expanded` gives:***
```python

loss = tensor([
        [1., 4., 9., 9., 9., 9.],
        [0., 0., 0., 0., 0., 0.],
        [1., 4., 9., 9., 9., 9.],
        [1., 4., 9., 9., 9., 9.],
        [1., 4., 9., 9., 9., 9.],
        [0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0.]
]) # sum = 164, numel = 48

torch.mean(loss) gives 3.4167.
```

**Using the wrong implementation: `torch.sum(squared_loss) / torch.sum(sample_weights)`**
```python

loss = F.mse_loss(tensor1, tensor2, reduction='none') 
loss = torch.tensor([
        [1., 4., 9., 9., 9., 9.],
        [0., 0., 0., 0., 0., 0.],
        [1., 4., 9., 9., 9., 9.],
        [0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0.]
]) # sum = 82, numel = 30

weighted_loss = loss * weight = torch.tensor([
        [ 1.,  4.,  9.,  9.,  9.,  9.],
        [ 0.,  0.,  0.,  0.,  0.,  0.],
        [ 3., 12., 27., 27., 27., 27.],
        [ 0.,  0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.,  0.]
]) # sum = 164, numel = 30

torch.sum(squared_loss * sample_weight) / torch.sum(sample_weights) gives 164/ 8 = 20.5
```

The results is therefore wrong, because we forgot to scale the weights by `dim1`. This PR fixes the bug by dividing the weighted loss by `torch.sum(sample_weights) * y_pred.shape[1].

Also ensures that tensors are either 1D or 2D. If 1D, reshape to 2D so `y_pred.shape[1]` works. Raises error if dimensions >2.

### CODE REVIEW

**Review by charlotte-avery — APPROVED** [review]
lgtm
